lib/view_model/active_record/association_manipulation.rb

```diff
@@ -0,0 +1,288 @@
+    association_data = self.class._association_data(association_name)
+    direct_reflection = association_data.direct_reflection
+    raise ArgumentError.new("Cannot append to single association '#{association_name}'") unless association_data.collection?
+
+    return_array     = subtree_hashes.is_a?(Array)
+    subtree_hashes   = Array.wrap(subtree_hashes)
+
+    model_class.transaction do
+      editable!(deserialize_context: deserialize_context, changed_associations: [association_name])
+
+      if association_data.through?
+        raise ArgumentError.new("Polymorphic through relationships not supported yet") if association_data.polymorphic?
+
+        direct_viewmodel_class = association_data.direct_viewmodel
+        root_update_data, referenced_update_data = construct_indirect_append_updates(association_data, subtree_hashes, references)
+      else
+        direct_viewmodel_class = association_data.viewmodel_class
+        root_update_data, referenced_update_data = construct_direct_append_updates(association_data, subtree_hashes, references)
+      end
+
+      update_context = UpdateContext.build!(root_update_data, referenced_update_data, root_type: direct_viewmodel_class)
+
+      # Provide information about what was updated
+      deserialize_context.updated_associations = root_update_data
+                                                   .map { |upd| upd.updated_associations }
+                                                   .inject({}) { |acc, assocs| acc.deep_merge(assocs) }
+
+      # Set new parent
+      new_parent = ViewModel::ActiveRecord::UpdateOperation::ParentData.new(direct_reflection.inverse_of, self)
+      update_context.root_updates.each { |update| update.reparent_to = new_parent }
+
+      # Set place in list.
+      if direct_viewmodel_class._list_member?
+        new_positions = select_append_positions(association_data,
+                                                direct_viewmodel_class._list_attribute_name,
+                                                update_context.root_updates.count,
+                                                before: before, after: after)
+
+        update_context.root_updates.zip(new_positions).each do |update, new_pos|
+          update.reposition_to = new_pos
+        end
+      end
+
+      updated_viewmodels = update_context.run!(deserialize_context: deserialize_context)
+
+      if association_data.through?
+        updated_viewmodels.map! do |direct_vm|
+          direct_vm._read_association(association_data.indirect_reflection.name)
+        end
+      end
+
+      if return_array
+        updated_viewmodels
+      else
+        updated_viewmodels.first
+      end
+    end
+  end
+
+  # Removes the association between the models represented by this viewmodel and
+  # the provided associated viewmodel. The associated model will be
+  # garbage-collected if the assocation is specified with `dependent: :destroy`
+  # or `:delete_all`
```

Comment:
No, this is now just wrong. Works for `through:`, but no longer for direct associations.